raiden/network/channelgraph.py

```diff
@@ -236,21 +235,32 @@ def __ne__(self, other):
         return not self.__eq__(other)
 
     def add_channel(self, details):
+        """ Instantiate a channel this node participates and add to the graph.
+
+        If the channel is already registered do nothing.
+        """
         channel_address = details.channel_address
-        partner_state = details.partner_state
-
-        channel = Channel(
-            details.our_state,
-            partner_state,
-            details.external_state,
-            self.token_address,
-            details.reveal_timeout,
-            details.settle_timeout,
-        )
-        self.add_path(details.our_state.address, partner_state.address)
-
-        self.partneraddress_to_channel[partner_state.address] = channel
-        self.address_to_channel[channel_address] = channel
+
+        if details.our_state.address != self.our_address:
```

Comment:
Shouldn't this code path be an assertion? If this happens it means that `add_channel` was called with completely wrong details and thus it's a programming error.

Reply:
I use assertion for code invariants, this is not an invariant but an invalid input

Reply:
Fair enough. Looks good apart from that. Thanks for the PR.